### PR TITLE
Implemented the CSV logic 

### PR DESCRIPTION
**Pull request by AymenBerb**
Alright I am going to do a PR now so that you guys can follow along easily. 

Until now, I created a CSV file named 'answers' that stores the selections of each tutor whenever they select their dates.

When a tutor submit its selection, the tutor record in the CSV file gets deleted as requested. I didn't implement yet the POST request to the API, since I am waiting for @ceciliacagnetta to finish T10.

There are still a few problems that I am aware of but are closely related:

1. Whenever there are two client announcements, if a tutor select his dates for both announcements without submitting after each selection, the answer to the first announcement will get deleted in other words, the CSV file will store only the tutor's latest selection. And so, if the tutor clicks on any submit button (can it be the one of the first or the second announcement) the bot will send the tutor's latest selection.
2. Same configuration as 1., but this time if the tutor clicks on both submit requests button, one of them will send his latest answer and the other will throw an error cause it will try to send data that has beed deleted.

So here is the current state of T09, if you guys have any questions, remarks etc.. Please lmk.

### CODE REVIEW

**Comment by AymenBerb** [comment]
So the version with Anton's solution is implemented. Please guys, I ask you to test it and see if everything works well and lmk if you have any recommendation to improve the UX. @absirgue @Ali-Alkhars

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
Very nice UI!! Amazing stuff. I think you forgot two things though:

1. Need to add a check if the tutor has anything selected before sending a "request sent" confirmation. To test, try to press submit without selecting anything.
2. Need to add a check if the tutor has anything selected before sending a "request cancelled" confirmation. To test, try to press cancel without selecting anything.

Another suggestion to enhance the UX, in the confirmation messages, could you add the selection that they confirmed/cancelled? For example "Your request for Monday 2PM has been sent." or "Your request for Tuesday 3PM has been cancelled".

File: index.js [hunk 1]
```diff
@@ -28,10 +37,184 @@ for (const file of commandFiles) {
...
+client.on('interactionCreate', async interaction => {
+	
+	// The answers of all tutors that are stored inside the CSV in the form of an array
+	const answers = await csv().fromFile("answers.csv");
+
+	if (interaction.customId === 'dateSelection') {
+	
+		// Checks if the tutor has already an answer stored inside then CVS
+		if (tutorMadeASelection(interaction.user.id, answers)) {
+			interaction.reply({ content: "Please **submit** or **cancel** your first request to proceed.", ephemeral: true });
```
Comment (absirgue) [inline index.js]: Excellent logic, I would just change the text to "You have registered answers to a previous message which you have not yet submitted or canceled. Please do so before attempting to register new one for this message." I mean that text is not that good but the idea is to say something that explains a bit better bcs the request they left answers for may be far up in the channel and they have registered answers a long time ago

File: index.js [hunk 2]
```diff
@@ -28,10 +37,184 @@ for (const file of commandFiles) {
...
+	if (interaction.customId === 'submitButton') {
+
+		if (tutorMadeASelection(interaction.user.id, answers)) {
+
+			// POST request to API is created with tutorId and selection under tutorDemand route
+
+			interaction.reply({content: "Your request has been sent.", ephemeral: true});
+
+			// Delete the appropriate line in the CSV and write the new CSV state
+			fsCsv.writeFileSync("answers.csv", new Parser({fields: [ "tutorId", "selection"] }).parse(deleteTutorAnswer(interaction.user.id, answers)));
```
Comment (absirgue) [inline index.js]: where are the answers read from the csv?

**Review by absirgue — COMMENTED** [review]
Really amazing stuff, thanks a ton!

Just one detail to fix and one thing I couldn't figure out but it's perfect. Congrats!

File: index.js (hunk 2 above)
Comment (AymenBerb) [reply index.js:143]: The deleteTutor function returns an array storing the updated state of the answers in the form of a list. This list is then written to the CSV.

**Review by absirgue — COMMENTED** [review]
Excellent excellent work. Just 2 comments:
- can you make the message saying that the tutor already has answers recorded for an other announcement standout a lil more by a color switch or bold text or something to make sure it's noticed.
- can you make it so that the cdv is automatically cleared when the server is started to avoid any conflict.
Thanks!

File: index.js [hunk 3]
```diff
@@ -29,9 +52,253 @@ for (const file of commandFiles) {
...
 client.once('ready', () => {
-	console.log('Ready!');
+	console.log('Ready !');
+	
+	//Delete every row in CSV
+	fsCsv.writeFileSync("answers.csv", new Parser({
+		fields: ["announcementId", "tutorId", "selection"]
+	}).parse([]));
+
+	sendNewClientMessage(1, ["Monday 9AM", "Wednesday 2PM", "Thursday 6PM"], 10, "Maths", "GCSE", 2, 1);
```
Comment (absirgue) [inline index.js]: I thought there would be a call to your API endpoint to get new course requests but I can't find it. If there is not, can u put in it or make the structure of those objects follow the actual structure of the object returned by the APU (with date options ID).

**Review by absirgue — COMMENTED** [review]
Hey, 

great work on this ticket.
Just a comment act connecting to the API and I also have another question: when I tested it out the result I selected did not log on the console so I couldn't see it really. Does that mean I have an issue with the version I am running or is it normal ?

THanks!

**Comment by Ali-Alkhars** [comment]
@AymenBerb please add any missing comments in the following files:
- index.js
- course-request.js
- messageUpdate.js

**Review by Ali-Alkhars — APPROVED** [review]
Amazing Aymen! I think you made it!

One thing I forgot to add though. Add the following to the course request confirmation message so it becomes the following:
```
            const courseRequestConfirmationMessage = `Your request for **${answer.replace(",", ", ")}** has been sent.\nYou will recieve an email about the status of your request when it is completed.`;
```

Merge approved🦊🦊
pop it

**Review by absirgue — APPROVED** [review]
Truly fantastic work Aymen.

a lot of efforts put in and it was definitely worth it. This is perfect!

thanks for the spirit! 🦊

**Pull request merged**